### DISCUSSION TITLE
Citar org-cite support merged to Doom biblio module

### DISCUSSION THREAD

**Discussion by bdarcus** [emacs-citar/citar#430] [General]
Just a head's up that org-cite support for the Doom Emacs biblio module is now merged.

Currently works best with the new, and now default, vertico completion module, which uses my citar package.

So basically, make sure the following are uncommented in your init.el file:

- biblio
- vertico
- org

... and then set the paths as noted in the module README, and you should be good to go.

https://github.com/hlissner/doom-emacs/tree/develop/modules/tools/biblio

This is the thread for questions and comments.

**Comment by wo** [comment]
Thanks for this! 

I've encountered one problem. When I open an org document with several nearby citations in org-cite format, most of them are not properly colour-coded as citations (indicating whether a reference is in the references database or not), but simply rendered as plain text. For example, when I open an org file with this content, only the first [cite...] element is coloured:

`Here are some citations: [cite:@williamson00knowledge]
Here are more: [cite:@adler2002belief]
[cite:@meacham2008sleeping] and
[cite:@hedden2015timeslice]`

When I move the cursor over one of the uncoloured entries and call embark, I get the citar menu.

Not sure if this is a bug (and if so, in what package) or a problem in my configuration, which is [here](https://github.com/wo/emacs-config).

**Comment by bdarcus** [comment]
I see the same thing, now that you mention it.

I think this is a bug in org, but it's an odd one.

Any chance you can test this by changing your activate processor to `basic` and see if you can reproduce?

Since the citar activate processor just uses that, you should get the same behavior, but it would help to know for a possible bug report to org.

EDIT: Hmm ... can you experiment more with this, using basic and citar activate processors? I find the behavior kind of flaky and inconsistent. 

For example, when I did this, I see the problem:

```elisp
Here's a citation [cite:@agnew2003], and another citation [cite:@couper2014]
```

But if I add a comma at the very end, that citation is then fontified, and inserting subsequent citations works as expected, so that in effect it's just a temporary problem.

**Comment by bdarcus** [comment]
BTW, on your config, you only need to set the citar paths. The module should handle the rest.

**Comment by wo** [comment]
The problem also happens if I set org-cite-activate-processor to basic. Oddly, if I change the setting while emacs is running (to basic or to citar) and then reload the file all citations are displayed properly. After restarting emacs the problem re-appears.

(Thanks for the hint about my config.)

**Comment by bdarcus** [comment]
I guess next step is to test it with emacs -q, or a non-Doom setup.

@localauthor @roshanshariff - do you see this behavior, such that we can figure out how to report it?

**Comment by roshanshariff** [comment]
I can replicate this behaviour with two citations on one line, but only when the first citation is a valid reference key:
```org
[cite:@Sno1959:TwoCultures] [cite:@Snow]
```
Just changing one character in the first citation key causes the second to flip between being highlighted or not.

**Reply by bdarcus** [reply]
So in your first citation, which yields correct behavior: both correct keys, or the first incorrect?

**Reply by roshanshariff** [reply]
When the first key is correct, the second is not highlighted, and vice versa.

**Comment by bdarcus** [comment]
From Nicolas:

> I couldn't reproduce it, but I saw something fishy in the fontification code. I fixed it. Hopefully, your problem is gone, too. Crossing fingers.

Does unpinning org work in Doom??

If we can confirm this works, maybe we can get it bumped.

**Reply by roshanshariff** [reply]
I have what I think is this commit (b3cc2f793, the latest one as of right now) and the above bug still happens for me.

**Reply by bdarcus** [reply]
I can reproduce this too, and have reported it. Hopefully that can help narrow it down.

**Reply by bdarcus** [reply]
He was able to reproduce, and merge a fix.

**Reply by bdarcus** [reply]
Just confirming, @wo, it's fixed in org. I assume the org module will be bumped soon, or you can unpin it.

**Reply by wo** [reply]
great, thanks for reporting! Really happy with this package btw. I never really understood all the ingredients in org-ref and orb, and things feel snappier now with the reduced weight.

**Reply by bdarcus** [reply]
ORB alone has equivalent LOC as citar, and org-ref has like 6x as much as either. And that's not considering the different dependencies (bibtex-completion etc).

This has less features, but I'm thinking that's a good thing for many.
